src/mbgl/gl/object_store.hpp

```diff
@@ -75,8 +66,13 @@ class ObjectStore : private util::noncopyable {
     }
 
     UniqueTexture createTexture() {
-        GLuint id = 0;
-        MBGL_CHECK_ERROR(glGenTextures(1, &id));
+        if (pooledTextures.empty()) {
+            pooledTextures.resize(TextureMax);
+            MBGL_CHECK_ERROR(glGenTextures(TextureMax, pooledTextures.data()));
+        }
+
+        GLuint id = pooledTextures.back();
```

Comment:
Got it now 👍 Indeed, after re-reading this patch I guess we didn't need to keep track of previous generated pools as we did before.